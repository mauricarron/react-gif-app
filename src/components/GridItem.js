import PropTypes from "prop-types";
const GridItem = ({ title, url }) => {
  return (
    <li className="card animate__animated animate__fadeIn">
      <img src={url} alt={title}></img>
    </li>
  );
};

GridItem.propTypes = {
  url: PropTypes.string.isRequired,
};

export default GridItem;
