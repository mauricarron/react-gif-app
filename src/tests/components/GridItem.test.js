import { shallow } from "enzyme";
import GridItem from "../../components/GridItem.js";

describe("tests on <GridItem />", () => {
  const title = "My Image";
  const url = "https://localhost/imagen.jpg";

  const wrapper = shallow(<GridItem title={title} url={url} />);

  test("should match component snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should match attributes in <img>", () => {
    const img = wrapper.find("img");

    expect(img.props().src).toBe(url);
    expect(img.props().alt).toBe(title);
  });

  test("should match css animation attributes ", () => {
    const animationAttribute = "animate__fadeIn";
    const li = wrapper.find("li");

    expect(li.prop("className")).toContain(animationAttribute);
  });
});
