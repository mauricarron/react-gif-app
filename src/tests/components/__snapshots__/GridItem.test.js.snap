// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <GridItem /> should match component snapshot 1`] = `
<li
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="My Image"
    src="https://localhost/imagen.jpg"
  />
</li>
`;
